### PR TITLE
Test config tweaks

### PR DESCRIPTION
**Pull request by ghost**
cache bundler for dev env (based on ENV variable), use ruby for tester config, etc

### CODE REVIEW

File: lib/jackal-kitchen/tester.rb [hunk 1]
```diff
@@ -27,8 +30,9 @@ def valid?(msg)
       #
       # @param msg [Carnivore::Message]
       def execute(msg)
+        dev_env = ENV['JACKAL_TEST'] == 'true'
         failure_wrap(msg) do |payload|
-          working_dir = working_path = Dir.mktmpdir
+          working_path = dev_env ? FileUtils.mkdir_p('/tmp/jackal-kitchen/bundle').first : Dir.mktmpdir
```
Comment (agoddard) [inline lib/jackal-kitchen/tester.rb]: move to config[:kitchen][:working_directory]

File: lib/jackal-kitchen/tester.rb [hunk 2]
```diff
@@ -42,12 +46,6 @@ def execute(msg)
...
               insert_kitchen_local(working_path) unless ENV['JACKAL_DISABLE_LXC']
-              netrc_file = File.open(File.expand_path('~/.netrc'), 'w')
-              gh_token = config.fetch(:github, :access_token,
-                app_config.get(:github, :access_token)
-              )
-              netrc_file.write("machine github.com\n  login #{gh_token}\n  password x-oauth-basic")
-              netrc_file.close
 
               run_commands(
                 [
```
Comment (agoddard) [inline lib/jackal-kitchen/tester.rb:52]: use config[:bundle_vendor_directory] here, fall back to `File.join(working_path, '.jackal-kitchen-vendor')`

File: lib/jackal-kitchen/tester.rb (hunk 1 above)
Comment (chrisroberts) [inline lib/jackal-kitchen/tester.rb]:
``` ruby
config[:working_directory]
```

Comment (chrisroberts) [inline lib/jackal-kitchen/tester.rb]: Also, nice working directory pattern: https://github.com/carnivore-rb/jackal-code-fetcher/blob/master/lib/jackal-code-fetcher/git_hub.rb#L15-L21

**Comment by cwjohnston** [comment]
:+1:

**Pull request merged**